src/test/scala/org/scanet/core/TFSpec.scala

```diff
@@ -0,0 +1,32 @@
+package org.scanet.core
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scanet.core.Session.using
+import org.scanet.core.Tensor.scalar
+import org.scanet.test.CustomMatchers
+import org.scanet.math.syntax._
+
+class TFSpec extends AnyFlatSpec with CustomMatchers {
+
+  "tensor function composition" should "work" in {
+    val sqr: TF1[Int, Int] = TF1 {
+      val arg = placeholder[Int]()
```

Comment:
would it make sense to create placeholders automatically as part of compile? (though in this case it might be problematic to assign label for the placeholder...)

Reply:
That was a demo sample. In reality, the placeholder will be compiled when the function is called the first time with the shape of the passed tensor. I think we also add alias feature.